common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1743,6 +1744,26 @@ def resource_url(self, resource):
     def publish(self, block, event_type, event):
         pass
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        service = self._services.get(service_name)
+        if callable(service):
```

Comment:
@asadiqbal08 @cpennington -- so is the solution here to add a check for either "needs" or "wants", otherwise skip?